src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicy.ts

```diff
@@ -336,33 +301,20 @@ export default class VideoPriorityBasedPolicy implements VideoDownlinkBandwidthP
     };
     rates.targetDownlinkBitrate = this.determineTargetRate();
 
-    // calculate subscribe delay based on bandwidth increasing/decreasing
     const numberOfParticipants = this.subscribedReceiveSet.size();
-    const prevEstimated = this.prevDownlinkStats.bandwidthEstimateKbps;
-    const currEstimated = this.downlinkStats.bandwidthEstimateKbps;
-
-    if (currEstimated > prevEstimated) {
-      // if bw increases, we use recovery delay
-      this.timeBeforeAllowSubscribeMs = this.getSubscribeDelay(
-        NetworkEvent.Increase,
-        numberOfParticipants
-      );
-    } else if (currEstimated < prevEstimated) {
-      // if bw decreases, we use response delay
-      this.timeBeforeAllowSubscribeMs = this.getSubscribeDelay(
-        NetworkEvent.Decrease,
-        numberOfParticipants
-      );
-    }
+    const currentEstimated = this.downlinkStats.bandwidthEstimateKbps;
 
     // If no major changes then don't allow subscribes for the allowed amount of time
     if (
       noMajorChange &&
-      Date.now() - this.lastSubscribeTimestamp < this.timeBeforeAllowSubscribeMs
+      this.probeFailed &&
+      !this.videoPriorityBasedPolicyConfig.allowSubscribe(numberOfParticipants, currentEstimated)
     ) {
       return;
     }
 
+    this.probeFailed = false;
```

Comment:
Nit: This needs a comment. Sometimes things are self-explanatory, sometimes not. But also in a class with this much mutable state I always recommend going overkill with the comments.

Reply:
i.e. when I see this in this diff my thought is immediately 'why' :P